src/RegularizedMoments.cc

```diff
@@ -0,0 +1,689 @@
+}
+
+// Calculates and assembles the matrix of the second moment from its sub components
+MomentsModel::SecondMoment makeBeta(Moments const & Q, Moments const & W) {
+    double x = BetaX::computeValue(Q, W);
+    double xy = BetaXY::computeValue(Q, W);
+    double y = BetaY::computeValue(Q, W);
+
+    MomentsModel::SecondMoment beta;
+    
+    beta(0, 0) = x;
+    beta(1, 0) = xy;
+    beta(0, 1) = xy;
+    beta(1, 1) = y;
+
+    return beta;
+}
+
+// Calculates quantities used in calculating the beta derivatives. These quantities are used in multiple
+// places so are calculated once and passed where needed
+double makeBetaGradDivisor(Moments const & Q, Moments const & W) {
+     double bottom = std::pow(Q[4],4)+4*std::pow(Q[4],3)*W[4]+std::pow(W[4],4)-2*W[3]*std::pow(W[4],2)*W[5]+
+                        std::pow(Q[3],2)*std::pow(W[5],2)+std::pow(W[3],2)*std::pow(W[5],2)+
+                        2*(3*std::pow(W[4],2)-Q[3]*W[5]-W[3]*W[5])*std::pow(Q[4],2)+(std::pow(Q[3],2)+
+                        2*Q[3]*W[3]+std::pow(W[3],2))*std::pow(Q[5],2)-2*(std::pow(W[4],2)*W[5]-
+                        W[3]*std::pow(W[5],2))*Q[3]+4*(std::pow(W[4],3)-Q[3]*W[4]*W[5]-W[3]*W[4]*W[5])*Q[4]-
+                        2*((Q[3]+W[3])*std::pow(Q[4],2)+W[3]*std::pow(W[4],2)-std::pow(Q[3],2)*W[5]-
+                        std::pow(W[3],2)*W[5]+(std::pow(W[4],2)-2*W[3]*W[5])*Q[3]+2*(Q[3]*W[4]+W[3]*W[4])*
+                        Q[4])*Q[5];
+    return bottom;
+}
+
+// Calculates and returns the derivatives for the beta components
+auto makeBetaGrad(Moments const & Q, Moments const & W) {
+    double bottom = makeBetaGradDivisor(Q, W);
+    return std::make_tuple(BetaX::computeGradient(Q, W, bottom), BetaXY::computeGradient(Q, W, bottom),
+                           BetaY::computeGradient(Q, W, bottom));
+}
+
+// Tests if two Moments vectors are approximately equal to some tolerance
+bool approxEqual(Moments const & first, Moments const & second, double tol=1e-6){
+    for (size_t i = 0; i < 6; ++i){
+        if (abs(first(i, 0) - second(i, 0)) > tol) {
+            return false;
+        }
+    }
+    return true;
+}
+
+} //end anonymous
+
+// Test the various calculations done by objects in the anonymous namespace
+
+bool testAlphaX(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = AlphaX::computeValue(Q, W);
+    double muBottom, sigBottom;
+    std::tie(muBottom, sigBottom) = makeAlphaGradDivisors(Q, W);
+    Moments firstRes = AlphaX::computeGradient(Q, W, muBottom, sigBottom);
+    double zeroTruth = 4.00033545790003;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0.557195571955720,
+                  -0.00335457900033546,
+                  -0.00411214444247764,
+                  0.00843596158202444,
+                  -0.0000506394012127103;
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
```

Comment:
`return (abs(zeroTruth - zeroRes) < tol) && approxEqual(firstRes, firstTruth, tol)`? Or move the first test higher to prevent the assignment cost? Or just leave these as they are, since they are minor tests anyway.